File: WKSandbox/ViewController.m
//
//  ViewController.m
//  WKSandbox
//
//  Created by Mac on 19/07/2017.
//
//

#import "ViewController.h"
#import "HTTPServer.h"

@implementation ViewController
{
    HTTPServer *server;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSString *path = [NSBundle.mainBundle pathForResource:@"html" ofType:nil];
    server = [[HTTPServer alloc] init];
    [server setDocumentRoot:path];
    NSError *error = nil;
    if (![server start:&error]) {
        NSLog(@"Error starting HTTP server: %@", error.localizedDescription);
        return;
    }

    NSString *serverPath = [NSString stringWithFormat:@"http://localhost:%d", server.listeningPort];
    NSLog(@"Running HTTP server: %@", serverPath);

    NSURL *url = [[NSURL alloc] initWithString:serverPath];
    NSURLRequest *request = [[NSURLRequest alloc] initWithURL:url];
    [self.webView loadRequest:request];
    
    [self.webView.configuration.preferences setValue:@YES forKey:@"developerExtrasEnabled"];
    self.webView.UIDelegate = self;
}

- (void)dealloc {
    [server stop];
}

- (void)webView:(WKWebView *)webView runJavaScriptAlertPanelWithMessage:(NSString *)message initiatedByFrame:(WKFrameInfo *)frame completionHandler:(void (^)(void))completionHandler {
    NSLog(@"alert=%@", message);
    completionHandler();
}

@end

File: P5Sandbox/ViewController.swift
//
//  ViewController.swift
//  P5Sandbox
//
//  Created by Dmitry on 15/07/2017.
//  Copyright © 2017 Dmitry. All rights reserved.
//

import Cocoa
import WebKit

class ViewController: NSViewController {
    @IBOutlet var webView: WKWebView!
    
    let server = HttpServer()
    
    override func viewDidLoad() {
        super.viewDidLoad()

        startServer()
        loadGame()

        webView.configuration.preferences.setValue(true, forKey: "developerExtrasEnabled")
    }
    
    func startServer() {
        if let htmlPath = Bundle.main.path(forResource: "html", ofType: nil) {
            server["/game/:path"] = shareFilesFromDirectory(htmlPath)
            try? server.start()
        }
    }
    
    func loadGame() {
        if let port = try? server.port() {
            if let url = URL(string: "http://localhost:\(port)/game/index.html") {
                let request = URLRequest(url: url)
                webView.load(request)
            }
        }
    }
    
    override var representedObject: Any? {
        didSet {
            // Update the view, if already loaded.
        }
    }
}
